### PR TITLE
489 lesson simplification

### PR DESCRIPTION
**Pull request by megganeturner**
Addresses #489

Lee's PR has been merged so as long as your backend is on an updated `FT-Offlining` branch this will work (:

Simplifies the lesson flow by merging all lesson cards into one flow. Wagtail now sends through all card detail in one array (`lesson.content.cards`). Lesson content can come through in any order.

- Card data comes through and is rendered as one `cards` array
- When entering a lesson, users no longer see the three activity links (objective, content and test). Instead, users see a new start card showing lesson details
- Pagination starts with the first card and includes all cards until the end of the content
- Remove success/completion screens that currently exist at the end of "objectives" and "content" sections

### CODE REVIEW

**Comment by PeteCoward** [comment]
Can we not delete 2 of LessonObjecttive, LessonTest and just use LessonContent?

File: src/riot/Lesson/LessonObjective.riot.html [hunk 1]
```diff
@@ -4,7 +4,8 @@
         extraStyleClasses="objective"
         progressValue="{ state.cardNumber }"
         progressMax="{ state.cards.length }"
-        linkTo="{props.lesson.loc_hash}"
+        lesson="{ props.lesson }"
```
Comment (katevoss) [inline src/riot/Lesson/LessonObjective.riot.html:7]: Why did you add this? I don't think the LessonFrame uses this prop at all.

File: src/riot/Lesson/LessonContent.riot.html [hunk 2]
```diff
@@ -4,7 +4,8 @@
         extraStyleClasses="content"
         progressValue="{ state.cardNumber }"
         progressMax="{ state.cards.length }"
-        linkTo="{props.lesson.loc_hash}"
+        lesson="{ props.lesson }"
```
Comment (katevoss) [inline src/riot/Lesson/LessonContent.riot.html:7]: As before, I don't think this prop is used.

**Comment by megganeturner** [comment]
> Can we not delete 2 of LessonObjecttive, LessonTest and just use LessonContent?

Okay, looking at Lee's PR (https://github.com/catalpainternational/canoe-backend/pull/112/files) I can see the structure of the data is much different to what I was expecting. Converting this to a draft and I'll continue working on this with the proper data in the backend!

**Review by PeteCoward — COMMENTED** [review]
@sandroroux needs to see this for the change in how we treat completions!

**Review by PeteCoward — APPROVED** [review]
I'll approve but leave open for @katevoss to look at and hopefully merge.

File: src/scss/_lesson.scss [hunk 3]
```diff
@@ -425,6 +418,11 @@ TestSingleAnswer {
             }
         }
     }
+
+    &.fade:after {
+        @extend %faded-modal-background;
+    }
```
Comment (katevoss) [inline src/scss/_lesson.scss:424]: I think this fixes https://github.com/catalpainternational/canoe/issues/502 right?

File: src/riot/Components/BottomMenu.riot.html [hunk 4]
```diff
@@ -41,7 +41,7 @@
                 return inAppelflap();
             },
             showBottomMenu() {
-                if (this.props.page.type === 'lessonpage' && this.state.section) {
+                if (this.props.page.type === 'lessonpage') {
```
Comment (katevoss) [inline src/riot/Components/BottomMenu.riot.html:44]: Unintentional effect of this is the Login page has a bottom menu now:


I don't think it's necessarily a bad thing but you should confirm with design/product that they want it there.

File: src/js/actions/completion.js [hunk 5]
```diff
@@ -54,15 +54,9 @@ const isLaterThanCutoff = (completion) => {
...
-    const objectiveCompletion = lessonMap.get("objectives");
     const contentCompletion = lessonMap.get("content");
-    const testCompletion = lessonMap.get("test");
 
-    return (
-        isLaterThanCutoff(objectiveCompletion) &&
-        isLaterThanCutoff(contentCompletion) &&
-        isLaterThanCutoff(testCompletion)
-    );
+    return isLaterThanCutoff(contentCompletion);
```
Comment (katevoss) [inline src/js/actions/completion.js:59]: There's an issue in completions: when I login, at first I don't see any completed courses (though I have completed many). Once I refresh the page, then I see the completed courses. @PeteCoward might have an idea as to why this is.

File: src/riot/Lesson/LessonContent.riot.html [hunk 6]
```diff
@@ -1,11 +1,15 @@
 <LessonContent>
     <LessonFrame
         if="{ props.lesson.ready && state.cardNumber <= state.cards.length }"
-        extraStyleClasses="content"
+        extraStyleClasses="{state.card.tag.includes('Choice') ? 'test' : 'content'}"
```
Comment (katevoss) [inline src/riot/Lesson/LessonContent.riot.html:4]: This seems like a brittle conditional. At the least I'd recommend putting it in the JS so that there's one place to fix if the tag name changes. For example,
```suggestion
        extraStyleClasses="{isAQuestion() ? 'test' : 'content'}"
```

And in the <script>
```
isAQuestion() {
     return this.state.card.tag.includes('Choice');
}
```

File: src/riot/Lesson/LessonContent.riot.html [hunk 7]
```diff
@@ -20,6 +24,24 @@
         <ListCard if="{state.card.tag === 'list'}" card="{state.card}" />
         <PdfCard if="{state.card.tag === 'pdf'}" card="{state.card}" />
         <Html5Card if="{state.card.tag === 'html5'}" card="{state.card}" />
+        <template
+            if="{state.card.tag.includes('Choice')}">
```
Comment (katevoss) [inline src/riot/Lesson/LessonContent.riot.html:28]: Then this would be 
```suggestion
            if="{isAQuestion()}">
```

File: src/riot/Lesson/TestPopupModal.riot.html [hunk 8]
```diff
@@ -38,16 +38,18 @@ <h3>{ props.isQuestionComplete ? correctTitleMessage(props.selectedAnswer) : inc
                 try: () => gettext('Try again'),
             },
 
-            nextTestCard() {
+            nextCard() {
                 this.props.toggleModal(false, {});
                 window.scroll(0, 0);
-                location.hash = this.props.lesson.loc_hash + ":test:" + parseInt(this.props.cardNumber + 1);
+                const nextCard = this.props.lesson.content.cards[this.props.cardNumber];
```
Comment (katevoss) [inline src/riot/Lesson/TestPopupModal.riot.html:44]: I don't think the `nextCard` value is used.

File: src/riot/WagtailPages/CoursePage.riot.html [hunk 9]
```diff
@@ -13,7 +13,7 @@ <h2 id="coursePageTitle">{course.title}</h2>
 
             <h5 class="section-title">{ TRANSLATIONS.your_lessons() }</h5>
             <div class="card-container">
-                <p if="{ !course.lessons.length }">{ TRANSLATIONS.no_lessons() }</h5>
+                <h5 if="{ !course.lessons.length }">{ TRANSLATIONS.no_lessons() }</h5>
```
Comment (katevoss) [inline src/riot/WagtailPages/CoursePage.riot.html:16]: 🤦🏻‍♀️ merging is an art, not a science

File: src/riot/WagtailPages/LessonPage.riot.html [hunk 10]
```diff
@@ -1,55 +1,32 @@
...
-                    status="{moduleStatus(TEST)}"
-                ></Card>
-            </div>
-        </template>
+    <LessonFrame
+        if="{lesson.ready}"
+        extrastyleclasses="message-container"
+        linkTo="{lesson.course.loc_hash}"
+    >
+        <h5>Lesson</h5>
```
Comment (katevoss) [inline src/riot/WagtailPages/LessonPage.riot.html:7]: This needs to be added to the translations object so it can be translated into Tetun.

**Review by katevoss — APPROVED** [review]
Moderate issues:
* Card images on the TitleDescription are not showing up, see the suggested lines in `LessonPage`
* Completions are not visible when I login. I have to refresh the page to see which courses I've completed.

Minor issue:
* Somewhere (I think in the LessonFrame) there's an extra 5px (or so) in the calculation of the card height, so that I can scroll slightly down on every card
    before scroll:
    Screen Shot 2021-03-17 at 09 34 56

    after scroll:
    Screen Shot 2021-03-17 at 09 36 47

**Comment by megganeturner** [comment]
Thanks Kate! I'll go through these and get the fixes in in my next PR. Not sure what's going on with the completion stuff, I'll take a closer look today!

**Comment by PeteCoward** [comment]
completions let me worry about. I'll get on it soon
